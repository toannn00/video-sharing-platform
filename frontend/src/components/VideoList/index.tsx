import { useMemo } from "react";
import Video from "../../types/video.type";
import { Empty } from "antd";
import VideoItem from "../VideoItem";
import { useMediaQuery } from "react-responsive";

export const VideoList = ({ videoList }: { videoList: Video[] }) => {
  const isMobile = useMediaQuery({ maxWidth: 1280 });

  const containerStyle = useMemo(
    () => ({
      width: isMobile ? "100%" : "70%",
      margin: isMobile ? "20px auto" : "40px auto 0",
      padding: isMobile ? 10 : 30,
    }),
    [isMobile]
  );

  return (
    <div style={containerStyle}>
      {videoList.length > 0 ? (
        videoList.map((video, index) => (
          <VideoItem key={`${video.url}-${index}`} video={video} />
        ))
      ) : (
        <Empty />
      )}
    </div>
  );
};

export default VideoList;
