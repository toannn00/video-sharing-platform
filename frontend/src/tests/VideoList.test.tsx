import "@testing-library/jest-dom";
import ShallowRenderer from "react-test-renderer/shallow";
import { vi } from "vitest";
import { BrowserRouter } from "react-router-dom";
import VideoList from "../components/VideoList";

vi.mock("react-router-dom", () => ({
  useNavigate: () => vi.fn(),
  BrowserRouter: ({ children }: { children: React.ReactNode }) => children,
}));

const mockVideo = {
  title: "test",
  description: "test",
  url: "test",
  email: "test@gmail.com",
};

const mockVideos = [mockVideo];

describe("VideoItem", () => {
  test("should render and match the snapshot", () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BrowserRouter>
        <VideoList videoList={mockVideos} />
      </BrowserRouter>
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
