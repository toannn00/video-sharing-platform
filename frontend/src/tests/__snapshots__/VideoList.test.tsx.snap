// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VideoItem > should render and match the snapshot 1`] = `
<VideoList
  videoList={
    [
      {
        "description": "test",
        "email": "test@gmail.com",
        "title": "test",
        "url": "test",
      },
    ]
  }
/>
`;

exports[`VideoList > should render and match the snapshot 1`] = `
<VideoList
  loading={false}
  videoList={
    [
      {
        "description": "test",
        "email": "test@gmail.com",
        "title": "test",
        "url": "test",
      },
    ]
  }
/>
`;
